src/TestSuite/IntegrationTestTrait.php

```diff
@@ -508,16 +498,19 @@ protected function _sendRequest($url, $method, $data = []): void
                 $this->_requestSession->write('Flash', $this->_flashMessages);
             }
             $this->_response = $response;
-        } catch (PhpUnitError $e) {
-            throw $e;
-        } catch (DatabaseException $e) {
-            throw $e;
+        } catch (\PHPUnit\Framework\Error\Error $e) {
+            $this->_exception = $e;
+        } catch (\Cake\Database\Exception $e) {
+            $this->_exception = $e;
         } catch (LogicException $e) {
-            throw $e;
+            $this->_exception = $e;
         } catch (Throwable $e) {
             $this->_exception = $e;
-            // Simulate the global exception handler being invoked.
-            $this->_handleError($e);
+        }
+
+        // Simulate the global exception handler being invoked.
+        if ($this->_exception instanceof Throwable) {
```

Comment:
Isnt this a behavior change?
They all are instance of this class afaik. Before only in the last catch block this was invoked, now in every case, right?

Reply:
Yes, previously it was:

```php
try {
   // ...
   $this->_response = $response;
   } catch (PhpUnitError $e) {
      throw $e;
   } catch (DatabaseException $e) {
      throw $e;
   } catch (LogicException $e) {
      throw $e;
   } catch (Throwable $e) {
      $this->_exception = $e;
      // Simulate the global exception handler being invoked.
      $this->_handleError($e);
}
```

I've updated all throwables to be added to the exception property and then passed to the handleError() to handle the exception. Thoughts?